File: app/metrics/files_modified.py
from metrics.base import metric


@metric
def files_modified(graph):
    """Computes the number of files modified in a particular commit."""
    result = []
    for commit in graph.iterate_commits():
        files_modified_in_commit = len(graph.commit_files[commit])
        result.append(files_modified_in_commit)
    return result

File: app/metrics/files_modified.js
Metrics['files_modified'] = {
  label: "Files Modified",
  icon: 'fa fa-file',
  description: "The number of files modified in a particular commit, includes new and deleted files.",
  long_description: null,
  colors: [
    '#f7f4f9',
    '#e7e1ef',
    '#d4b9da',
    '#c994c7',
    '#df65b0',
    '#e7298a',
    '#ce1256',
    '#91003f'
  ],
  mapper: Mappers['fibonacci_range'](),
  tooltip: function (value) {
    switch (value) {
      case 0:
        return "No files were modified";

      case 1:
        return "1 file was modified";

      default:
        return value + " files were modified";
    }
  }
};
